server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -550,6 +553,7 @@ static class LucenePersistedState implements PersistedState {
             // out by this version of OpenSearch. TODO TBD should we avoid indexing when possible?
             final PersistedClusterStateService.Writer writer = persistedClusterStateService.createWriter();
             try {
+                // With Remote State we can write cluster state with non empty Metadata but UNKNOWN_UUID cluster uuid
```

Comment:
During remote state restore, there will be non empty metadata getting persisted with cluster UUID as ClusterState.UNKOWN_UUID . The valid UUID will be generated and persisted along with the first cluster state getting published.